server/fleet/users.go

```diff
@@ -32,6 +32,26 @@ type User struct {
 	Teams []UserTeam `json:"teams"`
 }
 
+// IsGlobalObserver returns true if user is either a Global Observer or a Global Observer+
+func (u *User) IsGlobalObserver() bool {
+	if u.GlobalRole == nil {
+		return false
+	}
+	return *u.GlobalRole == RoleObserver || *u.GlobalRole == RoleObserverPlus
+}
+
+// TeamMembership returns a map whose keys are the TeamIDs of the teams for which pred evaluates to true
+// user is an Observer or Observer+
+func (u *User) TeamMembership(pred func(UserTeam) bool) map[uint]bool {
+	result := make(map[uint]bool, len(u.Teams))
```

Comment:
`result` is still an empty map even if initialized with a size, is this intended?